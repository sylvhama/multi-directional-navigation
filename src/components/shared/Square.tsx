import React from "react";
import styled from "styled-components";

import { Base } from "./Base";
import { Props } from "./types";

const StyledBase = styled(Base)`
  &:after {
    content: "";
    display: block;
    padding-bottom: 100%;
  }
`;

export const Square = React.forwardRef<HTMLDivElement, Props>(
  ({ children, isFocused, ...restProps }, ref) => {
    return (
      <StyledBase ref={ref} isFocused={isFocused} {...restProps}>
        {children}
      </StyledBase>
    );
  }
);
