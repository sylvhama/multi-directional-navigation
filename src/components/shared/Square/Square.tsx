import React from "react";
import styled from "styled-components";

import { Base } from "components/shared";
import { Props } from "components/shared/types";

const StyledBase = styled(Base)`
  &:after {
    content: "";
    display: block;
    padding-bottom: 100%;
  }
`;

export const Square = React.forwardRef<HTMLDivElement, Props>(
  ({ children, isFocused, ...restProps }, ref) => {
    return (
      <StyledBase ref={ref} isFocused={isFocused} {...restProps}>
        {children}
      </StyledBase>
    );
  }
);
