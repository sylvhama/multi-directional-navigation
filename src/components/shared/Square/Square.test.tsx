import "jest-styled-components";
import React from "react";
import { render } from "@testing-library/react";

import { Square } from "./Square";

describe.each([true, false])("<Square />", (isFocused) => {
  it(`renders style when isFocused is ${isFocused}`, () => {
    const { container } = render(<Square isFocused={isFocused} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
