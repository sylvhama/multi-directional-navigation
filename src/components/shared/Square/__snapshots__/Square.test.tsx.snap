// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Square /> renders style when isFocused is false 1`] = `
.c0 {
  border: 5px solid #333;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 4rem;
  background-color: #dfe3e8;
  color: #212b36;
}

.c1:after {
  content: "";
  display: block;
  padding-bottom: 100%;
}

<article
  class="c0 c1"
/>
`;

exports[`<Square /> renders style when isFocused is true 1`] = `
.c0 {
  border: 5px solid #333;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 4rem;
  background-color: #dfe3e8;
  color: #212b36;
  border-color: #006FBB;
  background-color: #F9FAFB;
  font-weight: bold;
}

.c1:after {
  content: "";
  display: block;
  padding-bottom: 100%;
}

<article
  class="c0 c1"
/>
`;
